src/app/portlets/dot-edit-page/content/services/dot-page-state/dot-page-state.service.ts

```diff
@@ -245,17 +244,18 @@ export class DotPageStateService {
         return of(null);
     }
 
-    private handleSetPageStateFailed(err: ResponseView): void {
-        this.dotHttpErrorManagerService
-            .handle(err)
-            .pipe(take(1))
-            .subscribe((res: DotHttpErrorHandled) => {
-                if (res.forbidden) {
+    private handleSetPageStateFailed(err: ResponseView): Observable<DotHttpErrorHandled> {
+        return this.dotHttpErrorManagerService.handle(err).pipe(
+            take(1),
+            tap(({ status }: DotHttpErrorHandled) => {
+                if (status === HttpCode.FORBIDDEN || status === HttpCode.NOT_FOUND) {
                     this.dotRouterService.goToSiteBrowser();
                 } else {
                     this.reload();
                 }
-            });
+            }),
+            map(() => null)
```

Comment:
what is this for?

Reply:
because the catchErrror neeeds to return something.